File: src/main/kotlin/io/titandata/titan/providers/kubernetes/Remove.kt
/*
 * Copyright (c) 2019 by Delphix. All rights reserved.
 */

package io.titandata.titan.providers.kubernetes

import io.titandata.client.apis.RepositoriesApi
import io.titandata.client.apis.VolumesApi
import io.titandata.titan.clients.Kubernetes

class Remove(
    private val kubernetes: Kubernetes = Kubernetes(),
    private val repositoriesApi: RepositoriesApi = RepositoriesApi(),
    private val volumeApi: VolumesApi = VolumesApi()
) {
    fun remove(repo: String, force: Boolean) {
        // TODO check running  & force
        // TODO why not working kubernetes.stopPortFowarding(repo)
        kubernetes.deleteStatefulSpec(repo)
        for (volume in volumeApi.listVolumes(repo)) {
            volumeApi.deleteVolume(repo, volume.name)
        }
        repositoriesApi.deleteRepository(repo)
        println("$repo removed")
    }
}

File: internal/app/providers/kubernetes/Remove.go
package kubernetes

import (
	"fmt"
	"strconv"
)

func Remove(repo string, force bool, port int) {
	cfg.BasePath = "http://localhost:" + strconv.Itoa(port)

	// TODO check running  & force
	// TODO why not working kubernetes.stopPortFowarding(repo)

	k8s.DeleteStatefulSpec(repo)
	vols, _, _ := volumesApi.ListVolumes(ctx, repo)
	for _, volume := range vols {
		volumesApi.DeleteVolume(ctx, repo, volume.Name)
	}
	repositoriesApi.DeleteRepository(ctx, repo)
	fmt.Println(repo + " removed")
}